lms/djangoapps/mobile_api/users/serializers.py

```diff
@@ -34,6 +35,10 @@ def to_representation(self, course_overview):
                 kwargs={'course_id': course_id},
                 request=request
             )
+            discussion_url = reverse(
+                'discussion_course',
+                kwargs={'course_id': course_id},
+                request=request) if course_overview.is_discussion_tab_enabled() else None
```

Comment:
The reason the `reverse` didn't work for you is because the URL is behind the discussion's feature flag: https://github.com/edx/edx-platform/blob/master/lms/urls.py#L494

So if you rearrange the code to the following, it should work with your changes in mobile_api/urls.py reverted:
```
if course_overview.is_discussion_tab_enabled():
    discussion_url = reverse(
        'discussion_course', 
        kwargs={'course_id': course_id}, 
        request=request,
    )
```

The preceding `if` statement will check that the Discussions feature flag is enabled.